test/functional/attribute_test.rb

```diff
@@ -0,0 +1,621 @@
+      assert_cmd(expected_result, result)
+    end
+
+    it "should print error on missing --compute-resource-id or --compute-resource." do
+      params = ['--interface', 'compute_name="eth1"', '--interface-id=1']
+      expected_result = usage_error_result(
+        @cmd,
+        "At least one of options --compute-profile-id, --compute-profile is required.",
+        "Could not update interface"
+      )
+
+      api_expects_no_call
+      result = run_cmd(@cmd + params)
+      assert_cmd(expected_result, result)
+    end
+
+    it "should update interface" do
+      params = ['--compute-profile-id=1', '--compute-resource-id=1', '--interface-id=1', '--interface', 'compute_name=eth0']
+
+      api_expects(:compute_profiles, :show) do |par|
+        par['id'] == 1
+      end.returns(@compute_profile)
+
+      api_expects(:compute_attributes, :update) do |par|
+        par['id'] == 2
+      end.returns(@compute_attribute)
+      result = run_cmd(@cmd + params)
+      assert_cmd(success_result("Interface was updated.\n"), result)
+    end
+  end
+
+  describe "remove-interface" do
+    before do
+      @cmd = ["compute-profile", "values", "remove-interface"]
+      @compute_profile =
+        {
+          "id" => 1,
+          "name" => "profile2",
+          "compute_attributes" => [{
+                                     "id" => 2,
+                                     "compute_resource_id" => 1,
+                                     "compute_resource_name" => "bla",
+                                     "provider_friendly_name" => "oVirt",
+                                     "compute_profile_id" => 1,
+                                     "compute_profile_name" => "profile2",
+                                     "vm_attrs" => {"interfaces_attributes" => { "1" => {"compute_name"=>"eth0"}}}
+                                   }]
+        }
+      @compute_attribute =  {
+        "id" => 2, "compute_resource_id" => 1, "compute_resource_name" => "bla",
+        "provider_friendly_name" => "oVirt", "compute_profile_id" => 1,
+        "compute_profile_name" => "profile2", "vm_attrs" => {"interfaces_attributes" => {}}
+      }
+
+    end
+
+    it "should print error on missing --compute-profile-id or --compute-profile" do
+      param = ['--interface-id=1']
+      expected_result = usage_error_result(
+        @cmd,
+        "At least one of options --compute-profile-id, --compute-profile is required.",
+        "Could not remove interface"
+      )
+
+      api_expects_no_call
+      result = run_cmd(@cmd + param)
+      assert_cmd(expected_result, result)
+    end
+
+    it "should print error on missing --compute-resource-id or --compute-resource" do
+      params = ['--compute-profile=A', '--interface-id=1']
+      expected_result = usage_error_result(
+        @cmd,
+        "At least one of options --compute-resource-id, --compute-resource is required.",
+        "Could not remove interface"
+      )
+
+      api_expects_no_call
```

Comment:
This test is failing as there is '--compute-profile=A' in the params which triggers name-resolution API call. Either change the param to '--compute-profile-id=1' (preferred) or add API call expectation to the test. 
The same pattern is causing failure in some other tests too.

Reply:
done